src/validation.cpp

```diff
@@ -904,7 +904,7 @@ bool MemPoolAccept::PreChecks(ATMPArgs& args, Workspace& ws)
         CAmount nDeltaFees = nModifiedFees - nConflictingFees;
         if (nDeltaFees < ::incrementalRelayFee.GetFee(nSize))
         {
-            return state.Invalid(TxValidationResult::TX_MEMPOOL_POLICY, "insufficient fee",
+            return state.Invalid(TxValidationResult::TX_MEMPOOL_POLICY, "insufficient replacement penalty",
```

Comment:
any reason to change the reject string? This will just cause a headache for downstream project (if they use the reject string) and might even be controversial due to the change "fee" -> "penalty".

Reply:
Also the commit title is wrong? Mentions rule 5, but this is rule 4?

Reply:
Pushed new commit, updating to rule 4.

W.r.t to headache for downstream project, it's picking up between the fact that this same reject reason is already used L897 for the "higher fee" check and L833 for the "higher feerate" check.

Do we want to dissociate the confusion ? I would say it's better but can just drop it if you think it's too bothering for downstream.

Reply:
If you really want to change it, a separate pull request with clear rationale to motivate the change (and the hassle it causes)  would be better. Bundling it with a change that is adding a test might not be the best approach.

Reply:
Obviously it needs release notes etc. Though, I'd rather avoid breaking downstream via reject reasons, which might break silently because they are not an enum. https://github.com/bitcoin/bitcoin/pull/19339#issuecomment-704894759

Reply:
Ah okay, well I'll let the confusion for now and just happy to have test coverage. Updated the branch.